src/Autofac.Extras.Moq/MoqRegistrationHandler.cs

```diff
@@ -82,6 +79,31 @@ public IEnumerable<IComponentRegistration> RegistrationsFor
             return new[] { rb.CreateRegistration() };
         }
 
+        private static bool CanMockService(IServiceWithType typedService)
+        {
+            return ServiceIsAbstractOrInterface(typedService) &&
+                   !IsGenericType(typedService) &&
+                   !IsIStartable(typedService);
+        }
+
+        private static bool IsIStartable(IServiceWithType typedService)
+        {
+            return typeof (IStartable).IsAssignableFrom(typedService.ServiceType);
+        }
+
+        private static bool IsGenericType(IServiceWithType typedService)

```

Comment:
Can we call this `IsUnsupportedGenericType`? If you read through the code it _appears from the name_ that you can't mock any generic type, which isn't true.